packages/viewer/src/app.ts

```diff
@@ -41,49 +41,88 @@ async function main(w: Window): Promise<void> {
 
   const document = w.document;
 
-  /**
-   * Parse the DOM looking for LaTeX and <math> elements.
-   * Replaces them with the corresponding rendered images within the given element.
-   * @param {Document} document - The DOM document in which to search for the rendering root.
-   * @param {MutationObserver} observer - Mutation observer to activate or reactivate every time the rendering root changes.
-   */
-  properties.render = async () => {
-    const element: HTMLElement = document.querySelector(properties.element);
-    if (element) {
-      await renderLatex(properties, element);
-      await renderMathML(properties, element);
-    }
-  };
-
   // Initial function to call once document is loaded
   // Renders formulas and sets observer
   const start = async () => {
-    // Check if the viewer is alredy loaded
+    // Check if the viewer is already loaded
     if (w.viewer.isLoaded) return;
 
     w.viewer.isLoaded = true;
 
-    // First render
-    properties.render();
+    // Interaction Observer
+    const mathobserver = new IntersectionObserver(async (entries, observer) => {
+
+      let latexPromises = [];
```

Comment:
I see that you are using the `simultaneousmml` but I fail to find the place where the `simultaneouslatex` is being used. It seems then that the parameter is not providing any value to the viewer at this point. Is this intentional, or am I failing to see something?

Reply:
Same here! 
In Latex.ts, line 21: 
```typescript
 const maxLatexPetition = properties.simultaneousmml;
```
Maybe maxLatexPetition should be equal to properties.simultaneouslatex? 

And in Mathml.ts, maybe we should use the same mechanism in the `renderMathML` function? 
or I am missing something?

Reply:
My mistake! It should be `const maxLatexPetition = properties.simultaneouslatex;`
FIxed, thank you both!